tests/drivers/i2s/i2s_api/src/main.c

```diff
@@ -28,80 +28,264 @@ void test_main(void)
 		k_object_access_grant(dev_i2s_tx, k_current_get());
 	}
 
-	ztest_test_suite(i2s_loopback_test,
-		ztest_unit_test(test_i2s_tx_transfer_configure_0),
-		ztest_unit_test(test_i2s_rx_transfer_configure_0),
-		ztest_unit_test(test_i2s_transfer_short),
-		ztest_unit_test(test_i2s_transfer_long),
-		ztest_unit_test(test_i2s_rx_sync_start),
-		ztest_unit_test(test_i2s_rx_empty_timeout),
-		ztest_unit_test(test_i2s_transfer_restart),
-		ztest_unit_test(test_i2s_transfer_rx_overrun),
-		ztest_unit_test(test_i2s_transfer_tx_underrun));
-	ztest_run_test_suite(i2s_loopback_test);
-
-	ztest_test_suite(i2s_states_test,
-		ztest_unit_test(test_i2s_tx_transfer_configure_1),
-		ztest_unit_test(test_i2s_rx_transfer_configure_1),
-		ztest_unit_test(test_i2s_state_not_ready_neg),
-		ztest_unit_test(test_i2s_state_ready_neg),
-		ztest_unit_test(test_i2s_state_running_neg),
-		ztest_unit_test(test_i2s_state_stopping_neg),
-		ztest_unit_test(test_i2s_state_error_neg));
-	ztest_run_test_suite(i2s_states_test);
-
-	ztest_test_suite(i2s_dir_both_loopback_test,
-		ztest_unit_test(test_i2s_dir_both_transfer_configure_0),
-		ztest_unit_test(test_i2s_dir_both_transfer_short),
-		ztest_unit_test(test_i2s_dir_both_transfer_long),
-		ztest_unit_test(test_i2s_dir_both_transfer_restart),
-		ztest_unit_test(test_i2s_dir_both_transfer_rx_overrun),
-		ztest_unit_test(test_i2s_dir_both_transfer_tx_underrun));
-	ztest_run_test_suite(i2s_dir_both_loopback_test);
-
-	ztest_test_suite(i2s_dir_both_states_test,
-		ztest_unit_test(test_i2s_dir_both_transfer_configure_1),
-		ztest_unit_test(test_i2s_dir_both_state_running_neg),
-		ztest_unit_test(test_i2s_dir_both_state_stopping_neg),
-		ztest_unit_test(test_i2s_dir_both_state_error_neg));
-	ztest_run_test_suite(i2s_dir_both_states_test);
-
-	/* Now run all tests in user mode */
```

Comment:
This is a rather strange way of testing user mode.

IMO, this should be handled through a separate userspace-enabled test (`drivers.i2s.userspace`) added to `testcase.yaml` that builds with `CONFIG_TEST_USERSPACE=y` -- that way, all these wrapper functions will not be necessary.